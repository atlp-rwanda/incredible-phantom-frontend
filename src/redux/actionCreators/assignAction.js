import axios from 'axios';
import { toast } from 'react-toastify';
import { BACKEND_URL } from '../../helpers/url';
import i18next from 'i18next';
import {
    GET_DRIVERS_FAILED,
    GET_DRIVERS_SUCCESS,
  
    
    ASSIGN_BUSDRIVER_FAILED,
    ASSIGN_BUSDRIVER_SUCCESS,
    
    UNASSIGN_BUSDRIVER_FAILED,
  
    UNASSIGN_BUSDRIVER_SUCCESS,

    GET_BUS_FAILED,
    GET_BUS_SUCCESS,
    
} from '../actionTypes/actionTypes';
import {token} from '../../components/AssignBusToDriver/token';
import { LaptopWindows } from '@material-ui/icons';

export const getDriversAction = () => async(dispatch) => {
    
    
    try {
        const response = await axios
        .get(`${BACKEND_URL}/api/users`,{
        headers: {
            auth: token
          }});


          dispatch({
            type: GET_DRIVERS_SUCCESS,
            payload: response.data.data
        })
    } catch (error) {
        dispatch({
            type: GET_DRIVERS_FAILED,
            payload: error.response.data.message
          });
    }

};


export const assignDrivertoBusAction = (dropDown, driverId) => async(dispatch) => { 
    try{
        const response = await axios({
            method: 'PATCH',
            url: `${BACKEND_URL}/api/users/${driverId}/assignToBus`,
            headers: {
             auth: token
            },
            data: {
             busId: dropDown
            }})
          
            
        response.status === 200 ? toast.success(i18next.t('assignDriver.success')) : null;
        location.reload();
        dispatch({
            type: ASSIGN_BUSDRIVER_SUCCESS,
            payload: response.data.message
        });
        
    } catch (error) {
        error.response.status === 500 ? toast.info(i18next.t('assignDriver.fail')) : null;
        dispatch({
            type: ASSIGN_BUSDRIVER_FAILED,
            payload: error.response
        })
    }
};

export const unAssignDriverToBusAction = (driverId) => async(dispatch) => {
 
    try{
        const response = await axios({
            method: 'PATCH',
            url: `${BACKEND_URL}/api/users/${driverId}/unassignToBus`,
            headers: {
             auth: token
            },
            })
        response.status === 200 ? toast.success(i18next.t('assignDriver.successU')) : null;
        location.reload();
        dispatch({
            type: UNASSIGN_BUSDRIVER_SUCCESS,
            payload: response.data.message
        });
    }
    catch(error){
        error.response.status === 500 ? toast.info(i18next.t('assignDriver.failU')) : null;
        dispatch({
            type: UNASSIGN_BUSDRIVER_FAILED,
            payload: error.response
        })
    }
};

export const getBusAction = () => async(dispatch) => { 
    try {
        const response = await axios
        .get(`${BACKEND_URL}/api/bus`,{
        headers: {
            auth: token
          }});


          dispatch({
            type: GET_BUS_SUCCESS,
            payload: response.data.data
        })
    } catch (error) {
        dispatch({
            type: GET_BUS_FAILED,
            payload: error.response.data.message
          });
    }
}
