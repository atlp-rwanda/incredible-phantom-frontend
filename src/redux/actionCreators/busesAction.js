import axios from 'axios';
import { toast } from 'react-toastify';
import i18next from 'i18next';
import { BACKEND_URL } from '../../helpers/url';
import {
  GETTING_BUSES,
  GETTING_BUSES_ERROR,
  GETTING_BUSES_SUCCESS,
  EDIT_BUS,
  EDIT_BUS_ERROR,
  EDIT_BUS_SUCCESS,
  DELETE_BUS,
  DELETE_BUS_ERROR,
  DELETE_BUS_SUCCESS,
  CREATE_BUS,
  CREATE_BUS_ERROR,
  CREATE_BUS_SUCCESS
} from '../actionTypes/actionTypes';

import { token } from '../../components/dashboard/buses/helpers';

export const getBusesAction = () => async (dispatch) => {
  dispatch({ type: GETTING_BUSES });
  try {
    const authHeader = `Bearer ${token}`;
    const response = await axios({
      method: 'GET',
      url: `${BACKEND_URL}/api/bus`,
      headers: {
        auth: authHeader
      }
    });
    let data = response.data.data;
    let sortedData = data.reverse();
    dispatch({
      type: GETTING_BUSES_SUCCESS,
      payload: sortedData
    });
  } catch (error) {
    dispatch({
      type: GETTING_BUSES_ERROR,
      payload: error.response
    });
  }
};

export const createBusAction = (newBus) => async (dispatch) => {
  dispatch({ type: CREATE_BUS });
  console.log('Data', newBus);
  try {
    const authHeader = `Bearer ${token}`;

    const response = await axios.post(
      `${BACKEND_URL}/api/bus`,

      newBus,
      {
        headers: {
          auth: authHeader
        }
      }
    );
    response.status === 201 ? toast.success(i18next.t('buses.created')) : null;
    dispatch({ type: CREATE_BUS_SUCCESS, payload: response.data.message });
  } catch (error) {
    console.log(error);
    error.response.status === 400
      ? toast.info(i18next.t('buses.already'))
      : null;
    dispatch({ type: CREATE_BUS_ERROR, payload: error.response });
  }
};

export const editBusAction = (input, busID) => async (dispatch) => {
  dispatch({ type: EDIT_BUS });
  console.log(input);
  try {
    const authHeader = `Bearer ${token}`;
    const response = await axios({
      method: 'PATCH',
      url: `${BACKEND_URL}/api/bus/${busID}`,
      headers: {
        auth: authHeader
      },
      data: {
        ...input
      }
    });
    dispatch({ type: EDIT_BUS_SUCCESS, payload: response.data.message });
  } catch (error) {
    error.response.status === 400 ? toast.info(error.response.message) : null;
    dispatch({ type: EDIT_BUS_ERROR, payload: error.response });
  }
};

export const deleteBusAction = (busID) => async (dispatch) => {
  dispatch({ type: DELETE_BUS });
  try {
    const authHeader = `Bearer ${token}`;
    const response = await axios({
      method: 'DELETE',
      url: `${BACKEND_URL}/api/bus/${busID}`,
      headers: {
        auth: authHeader
      }
    });
    dispatch({ type: DELETE_BUS_SUCCESS, payload: response.data.message });
  } catch (error) {
    error.response ? toast.info(error.response.message) : null;
    dispatch({ type: DELETE_BUS_ERROR, payload: error.response });
  }
};
