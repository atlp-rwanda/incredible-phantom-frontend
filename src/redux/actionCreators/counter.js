export const counterAction = (nbr) => ({ type: 'COUNTER', payload: nbr + 2 });
