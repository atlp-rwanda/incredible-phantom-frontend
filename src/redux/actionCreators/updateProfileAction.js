import axios from 'axios';
import i18next from 'i18next';
import { toast } from 'react-toastify';
import { BACKEND_TOKEN, BACKEND_URL } from '../../helpers/url';
import {
  UPDATE_PROFILE,
  UPDATE_SUCCESS,
  UPDATE_FAILED
} from '../actionTypes/profileType';

const userId = JSON.parse(localStorage.getItem('loggedInUser'));

export const updateProfileAction = (firstName, lastName, phone) => async (
  dispatch
) => {
  dispatch({ type: UPDATE_PROFILE });
  try {
    const res = await axios.patch(
      `${BACKEND_URL}/api/users/update/${userId.id}`,
      {
        firstName,
        lastName,
        phone
      },
      { headers: { auth: BACKEND_TOKEN } }
    );
    dispatch({ type: UPDATE_SUCCESS, payload: res.data.data });
  } catch (error) {
    dispatch({ type: UPDATE_FAILED });
    toast.error(i18next.t(`UPDATE_FAILED ${error?.response?.data?.message}`));
  }
};
