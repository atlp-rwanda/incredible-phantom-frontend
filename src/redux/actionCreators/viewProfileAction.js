import axios from 'axios';
import { BACKEND_TOKEN, BACKEND_URL} from '../../helpers/url';
import {
  VIEW_PROFILE,
  VIEW_SUCCESS,
  VIEW_FAIL
} from '../actionTypes/profileType';

const userId = JSON.parse(localStorage.getItem('loggedInUser'));
console.log(userId)
export const viewProfileAction = () => async (dispatch) => {
  
  try {
    dispatch({ type: VIEW_PROFILE });
    const res = await axios.get(`${BACKEND_URL}/api/users/${userId.id}`, {
      headers: {
        auth: BACKEND_TOKEN
      }
    });
    dispatch({ type: VIEW_SUCCESS, payload: res.data });
  } catch (error) {
    console.log(error.response);
    dispatch({ type: VIEW_FAIL, payload: error?.response?.message });
  }
};

