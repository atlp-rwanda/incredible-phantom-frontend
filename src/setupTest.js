import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

Enzyme.configure({
  adapter: new EnzymeAdapter(),
});

export const mockStore = configureMockStore([thunk]);
