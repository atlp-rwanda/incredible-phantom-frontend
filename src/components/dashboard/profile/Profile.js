import React, { Component } from 'react';
import '../../../styles/profile.scss';
import Modal from 'react-modal';
import 'react-toastify/dist/ReactToastify.css';
import Skeleton from 'react-loading-skeleton';
import { viewProfileAction } from '../../../redux/actionCreators/viewProfileAction';
import { updateProfileAction } from '../../../redux/actionCreators/updateProfileAction';
import { Trans } from 'react-i18next';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

class Profile extends Component {
  constructor() {
    super();
    this.state = {
      ModalIsOpen: false,
      firstName: '',
      lastName: '',
      phone: ''
    };
  }
  componentDidMount() {
    setInterval(() => {
      this.props.viewProfileAction();
    }, 2000);
  }
  handleInput = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };
  handleSubmit = () => {
    this.props.updateProfileAction(
      this.state.firstName,
      this.state.lastName,
      this.state.phone
    );
    if (updateProfileAction() !== true) {
      return { ModalIsOpen: true };
    }
  };
  render() {
    const { ModalIsOpen } = this.state;
    return this.props.profile ? (
      <div className='all'>
        <Modal
          className='popup1'
          isOpen={ModalIsOpen}
          style={{ overlay: { backgroundColor: 'rgba(0,0,0,0.25)' } }}
        >
          <div className='Edit-container-bg'>
            <div className='Edit-container'>
              <p className='pEdit'>
                {<Trans i18nKey='profile.editprofile'></Trans>}
              </p>
              <form
                data-testId='form1'
                onSubmit={(e) => {
                  e.preventDefault();
                  this.handleSubmit();
                  this.setState({ ...this.state, ModalIsOpen: false });
                }}
              >
                <ul className='edit-table'>
                  <div className='item'>
                    <p className='dataEdit'>
                      {<Trans i18nKey='profile.firstname'></Trans>}
                    </p>
                    <input
                      type='text'
                      data-testid='firstName'
                      className='keyEdit'
                      placeholder='first Name'
                      onChange={this.handleInput}
                      name='firstName'
                      defaultValue={this.props.profile.data.firstName}
                    />
                  </div>
                  <div className='item'>
                    <p className='dataEdit'>
                      {<Trans i18nKey='profile.lastname'></Trans>}
                    </p>
                    <input
                      type='text'
                      data-testid='lastName'
                      className='keyEdit'
                      placeholder='Last name'
                      onChange={this.handleInput}
                      name='lastName'
                      defaultValue={this.props.profile.data.lastName}
                    />
                  </div>
                  <div className='item'>
                    <p className='dataEdit'>
                      {<Trans i18nKey='profile.phonenumber'></Trans>}
                    </p>
                    <input
                      data-testid='phone'
                      type='number'
                      className='keyEdit'
                      placeholder='Phone Number'
                      onChange={this.handleInput}
                      name='phone'
                      defaultValue={this.props.profile.data.phone}
                    />
                  </div>
                </ul>
                <div className='error'></div>
                <button type='submit' className='save' data-testid='save'>
                  {<Trans i18nKey='profile.save'></Trans>}
                </button>
                <button
                  data-testId='cancel1'
                  type='button'
                  className='cancel'
                  onClick={() =>
                    this.setState({ ...this.state, ModalIsOpen: false })
                  }
                >
                  {<Trans i18nKey='profile.cancel'></Trans>}
                </button>
              </form>
            </div>
          </div>
        </Modal>
        <div className={`ProfilePage`}>
          <div className={`table`}>
            <div className='Container'>
              <h2> {<Trans i18nKey='profile.profile'></Trans>}</h2>
              <div className='data-container'>
                <label className='key'>
                  {<Trans i18nKey='profile.firstname'></Trans>}
                </label>
                <span className='data' id='data1' placeholder='First Name'>
                  {this.props?.profile?.data.firstName}
                </span>
              </div>
              <div className='data-container'>
                <label className='key'>
                  {<Trans i18nKey='profile.lastname'></Trans>}
                </label>
                <span className='data' placeholder='Last Name'>
                  {this.props?.profile?.data.lastName}
                </span>
              </div>
              <div className='data-container'>
                <label className='key'>
                  {<Trans i18nKey='profile.role'></Trans>}
                </label>
                <span className='data' placeholder='Role'>
                  {this.props?.profile?.data.role}
                </span>
              </div>
              <div className='data-container'>
                <label className='key'>
                  {<Trans i18nKey='profile.nationalid'></Trans>}
                </label>
                <span className='data' placeholder='National ID'>
                  {this.props?.profile?.data.nationalId}
                </span>
              </div>
              <div className='data-container'>
                <label className='key'>
                  {<Trans i18nKey='profile.email'></Trans>}
                </label>
                <span className='data' placeholder='Email'>
                  {this.props?.profile?.data.email}
                </span>
              </div>
              <div className='data-container'>
                <label className='key'>
                  {<Trans i18nKey='profile.phonenumber'></Trans>}
                </label>
                <span className='data' placeholder='Phone' id='phone'>
                  {this.props?.profile?.data.phone}
                </span>
              </div>
              <div className='buttons'>
                <button
                  data-testid='submit1'
                  type='submit'
                  className='edit'
                  onClick={() =>
                    this.setState({ ...this.state, ModalIsOpen: true })
                  }
                >
                  {<Trans i18nKey='profile.edit'></Trans>}
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    ) : (
      <Skeleton className='skel' />
    );
  }
}
const mapStateToProps = (state) => ({
  profile: state.viewProfileReducer.data
});
const mapDispatchToProps = (dispatch) => {
  return {
    viewProfileAction: () => dispatch(viewProfileAction()),
    updateProfileAction: (firstName, lastName, phone) =>
      dispatch(updateProfileAction(firstName, lastName, phone))
  };
};
export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Profile)
);
