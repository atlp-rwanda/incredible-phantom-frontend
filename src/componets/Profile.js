import React from 'react';
import '../styles/profile.scss';
//import {connect} from 'react-readux';



const Profile = ()=>{
    return (
        <div className="profile">
            <div>
            <img src="images.jpeg" alt=""/>
            <div>
            </div>
                <form className="form">
                    <div id="name">
                        <label>Names:</label>
                        <input type="text" name="Name" required></input>
                    </div>
                    <div id="NID">
                        <label>National ID:</label>
                        <input type="number" name="National Id" required></input>
                    </div>
                    <div id="phone">
                        <label>Phone:</label>
                        <input type="number" name="Phone" required></input>
                    </div>
                    <div id="email">
                        <label>Email:</label>
                        <input type="text" name="Email" required></input>
                    </div>
                    <div id="button">
                        <button className="Edit" id="edit">Edit</button>
                        <button className="Delete" id="delete">Delete</button> 
                    </div>
                </form>
            </div>
        </div>
    );

};

export default Profile;
