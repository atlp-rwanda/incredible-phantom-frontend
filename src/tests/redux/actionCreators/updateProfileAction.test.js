import React from 'react';
import { Provider } from 'react-redux';
import Profile from '../../../components/dashboard/profile/Profile';
import { shallow } from 'enzyme';
import moxios from 'moxios';
import axiosInstance from 'axios';
import { mockStore } from '../../../setupTest';
import {
  UPDATE_FAILED,
  UPDATE_SUCCESS,
  UPDATE_PROFILE,
  VIEW_FAIL,
  VIEW_PROFILE,
  VIEW_SUCCESS
} from '../../../redux/actionTypes/profileType';
import { viewProfileAction } from '../../../redux/actionCreators/viewProfileAction';
import { updateProfileAction } from '../../../redux/actionCreators/updateProfileAction';

const store = mockStore();
const flushPromises = () => new Promise((resolve) => setImmediate(resolve));

describe('profile', () => {
  it('should take a snapshot', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Profile />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('update profile Actions', () => {
  beforeEach(() => {
    moxios.install(axiosInstance);
  });

  afterEach(() => {
    moxios.uninstall(axiosInstance);
  });

  it('Should test updating profile loading ', async (done) => {
    moxios.wait(async () => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: {
          success: true,
          message: 'profile updated successfully',
          data: {}
        }
      });
    });
    await store.dispatch(updateProfileAction('test', 'test', 'test'));
    await flushPromises();
    const calledActions = store.getActions();
    expect(calledActions[0].type).toEqual(UPDATE_PROFILE);
    done();
  });

  it('Should test updating profile success ', async (done) => {
    moxios.wait(async () => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: {
          success: true,
          message: 'profile updated successfully',
          data: {}
        }
      });
    });
    await store.dispatch(updateProfileAction('sibo', 'jean', '82323233434'));
    await flushPromises();
    const calledActions = store.getActions();
    done();
  });

  it('Should fail updating user ', async (done) => {
    moxios.wait(async () => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 404,
        response: {
          success: false,
          message: 'error',
          data: {}
        }
      });
    });
    await store.dispatch(updateProfileAction());
    await flushPromises();
    const calledActions = store.getActions();
    expect(calledActions[1].type).toEqual(UPDATE_FAILED);
    done();
  });
});

describe('VIEW profile Actions', () => {
  beforeEach(() => {
    moxios.install(axiosInstance);
  });

  afterEach(() => {
    moxios.uninstall(axiosInstance);
  });

  it('Should test VIEW profile loading ', async (done) => {
    moxios.wait(async () => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: {
          success: true,
          message: 'profile Vewed successfully',
          data: {}
        }
      });
    });
    await store.dispatch(viewProfileAction());
    await flushPromises();
    const calledActions = store.getActions();
    done();
  });

  it('Should test VIEW profile success ', async (done) => {
    moxios.wait(async () => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: {
          success: true,
          message: 'profile Viewed successfully',
          data: {}
        }
      });
    });
    await store.dispatch(viewProfileAction());
    await flushPromises();
    const calledActions = store.getActions();
    done();
  });

  it('Should fail Viewing Profile user ', async (done) => {
    moxios.wait(async () => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 404,
        response: {
          success: false,
          message: 'error',
          data: {}
        }
      });
    });
    await store.dispatch(viewProfileAction());
    await flushPromises();
    const calledActions = store.getActions();
    done();
  });
});
