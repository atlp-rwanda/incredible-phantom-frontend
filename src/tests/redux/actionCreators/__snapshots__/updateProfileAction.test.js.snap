// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile should take a snapshot 1`] = `ShallowWrapper {}`;
